src/main/groovy/org/exoplatform/platform/am/utils/FileEncrypt.groovy

```diff
@@ -0,0 +1,105 @@
+package org.exoplatform.platform.am.utils
+
+import java.io.*
+import java.util.Base64
+import javax.crypto.Cipher
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.CipherInputStream
+import javax.crypto.CipherOutputStream
+import javax.crypto.spec.SecretKeySpec
+import java.security.spec.AlgorithmParameterSpec
+
+
+ class FileEncrypt {
+     Cipher ecipher
+     Cipher dcipher
+    private final String ENC_KEY = "noFUD4Zn8zNVdEtfxMcW5w==" // DO_NOT_CHANGE_IT!!!!
+
+     FileEncrypt() {
+
+        byte[] decodedKey = Base64.getDecoder().decode(ENC_KEY)
+        SecretKey key = new SecretKeySpec(decodedKey, 0, decodedKey.length, "AES")
+
+        // Create an 8-byte initialization vector
+        byte[] iv = [
+                0x00,
+                0x01,
+                0x02,
+                0x03,
+                0x04,
+                0x05,
+                0x06,
+                0x07,
+                0x08,
+                0x09,
+                0x0a,
+                0x0b,
+                0x0c,
+                0x0d,
+                0x0e,
+                0x0f
+     ]
+
+        AlgorithmParameterSpec paramSpec = new IvParameterSpec(iv)
+        try {
+            ecipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
+            dcipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
+
+            // CBC requires an initialization vector
+            ecipher.init(Cipher.ENCRYPT_MODE, key, paramSpec)
+            dcipher.init(Cipher.DECRYPT_MODE, key, paramSpec)
+        } catch (Exception e) {}
+    }
+
+    // Buffer used to transport the bytes from one stream to another
+    byte[] buf = new byte[1024]
+
+    private void encrypt(InputStream inp , OutputStream out) {
+        try {
+            // Bytes written to out will be encrypted
+            out = new CipherOutputStream(out, ecipher)
+
+            // Read in the cleartext bytes and write to out to encrypt
+            int numRead = 0
+            while ((numRead = inp .read(buf)) >= 0) {
+                out.write(buf, 0, numRead)
+            }
+            out.close()
+        } catch (java.io.IOException e) {}
```

Comment:
Plese log the exception

Reply:
Done

Reply:
The use of "Apply" command is restricted for supported customers. In order to ensure patches came from the support service. i have added the encryption step of the patch file. I had also prepared a tiny java (jar) program which encrypts / decrypts the patch file that will provided to our support service. In addition, IMO, there is no need to show clients what is in the encrypted File because it won't be necessary. However, while installing the patch the addons-manager will show what's happening (what jar / war are changing) . Also the CRC Verification would be helpful to check file tampering detection (Optional). These mechanisms are only available for the offline installation of eXo Patches.